lib/sinon.js

```diff
@@ -7,7 +7,21 @@
  * Copyright (c) 2010-2013 Christian Johansen
  */
 "use strict";
-module.exports = require("./sinon/util/core");
+
+var match = require("./sinon/match");
+
+module.exports = exports = require("./sinon/util/core");
+
+exports.extend = require("./sinon/extend");
+exports.match = match;
+exports.typeOf = require("./sinon/typeOf");
+
+/*
+ * allow deepEqual to check equality of matchers through
+ * dependency injectection. Otherwise we get a circular
+ * dependency
+ */
+exports.deepEqual.match = match;
```

Comment:
@mroderick Who would make that call? That's breaking backwords compatibility. `deepEqual(matcher, foo)` will change behavior. That's what I was trying to avoid.